apps/glusterfs/volume_entry.go

```diff
@@ -335,21 +355,71 @@ func ReduceRawSize(size int) int {
 // AddRawCapacity adds raw capacity to the BlockInfo
 // FreeSize tracking, reserving one percent of the
 // raw capacity for the filesystem.
-func (v *VolumeEntry) AddRawCapacity(delta int) {
+func (v *VolumeEntry) AddRawCapacity(delta int) error {
 	var freeDelta int
 	var reservedDelta int
 
 	freeDelta = ReduceRawSize(delta)
 	reservedDelta = delta - freeDelta
 
-	v.ModifyFreeSize(freeDelta)
-	v.ModifyReservedSize(reservedDelta)
+	if err := v.ModifyFreeSize(freeDelta); err != nil {
+		return err
+	}
+	if err := v.ModifyReservedSize(reservedDelta); err != nil {
+		return err
+	}
+	return nil
 }
 
 func (v *VolumeEntry) SetRawCapacity(delta int) {
 	v.Info.BlockInfo.FreeSize = 0
 	v.Info.BlockInfo.ReservedSize = 0
-	v.AddRawCapacity(delta)
+	if err := v.AddRawCapacity(delta); err != nil {
+		// if this fails, something is deeply wrong and panic is appropriate
```

Comment:
How is this any different from e.g. AddRawCapacity?
I.e., why do we need to treat this specially?

Reply:
SetRawCapacity should only be called when the volume is being created. It really should not be able to trigger the error conditions being checked because of the values being set to zero. Thus the "deeply wrong" comment.

If the asymmetry bugs you I can easily change it to return an error, the rarity/severity of it seemed like it was still more of a panic level thing.

Reply:
It is imho irrelevant that `FreeSize` has been set to 0, because the resulting FreeSize is checked against `volume.Info.Size`. So if you call `SetRawCapacity()` with something larger than `volume.Info.Size` it will fail.

In that regard, `SetRawCapacity()` is not different than any other call of `ModifyFreeSize()` or `ModifyReservedSize()`. Hence I would say either panic all or panic none.

Making any sense?

Reply:
Panic none it is then. I will implement as soon as I can.